### PR TITLE
Ignore test directories in coverage

### PR DESCRIPTION
**Pull request by j-rivero**
Test directories should not be included in the coverage reports.

Testing:
 * Original report for spdlog https://ci.ros2.org/job/ci_linux_coverage/140/cobertura/
 * Running this branch: [Build Status](https://ci.ros2.org/job/ci_linux_coverage/145/)

### CODE REVIEW

**Comment by j-rivero** [comment]
Trying new regexp [Build Status](https://ci.ros2.org/job/ci_linux_coverage/148/)

**Comment by j-rivero** [comment]
Yeah, I think we get it https://ci.ros2.org/job/ci_linux_coverage/153/cobertura/

**Comment by nuclearsandwich** [comment]
Should we also add an ignore for gtest_vendor https://ci.ros2.org/job/ci_linux_coverage/153/cobertura/install_gtest_vendor_src_gtest_vendor_include_gtest/

**Comment by nuclearsandwich** [comment]
> Should we also add an ignore for gtest_vendor [ci.ros2.org/job/ci_linux_coverage/153/cobertura/install_gtest_vendor_src_gtest_vendor_include_gtest](https://ci.ros2.org/job/ci_linux_coverage/153/cobertura/install_gtest_vendor_src_gtest_vendor_include_gtest/)

I added a commit which does exactly that: https://ci.ros2.org/job/ci_linux_coverage/163/cobertura/

**Comment by j-rivero** [comment]
> I added a commit which does exactly that: https://ci.ros2.org/job/ci_linux_coverage/163/cobertura/

thanks! seems to work to me.

File: ros2_batch_job/__main__.py [hunk 1]
```diff
@@ -298,7 +298,8 @@ def process_coverage(args, job):
         # cobertura plugin looks for files of the regex *coverage.xml
         outfile = os.path.join(package_build_path, package_name + '.coverage.xml')
         print('Writing coverage.xml report at path {}'.format(outfile))
-        # -e /usr  Ignore files from /usr
+        # --gcov-exlcude remove generated .gcov files from previous gcov call.
```
Comment (nuclearsandwich) [inline ros2_batch_job/__main__.py:301]:
```suggestion
        # --gcov-exclude remove generated .gcov files from previous gcov call. 
```

**Comment by j-rivero** [comment]
Skip tests directory too in the last commit, tested here https://ci.ros2.org/job/ci_linux_coverage/171/cobertura/

File: ros2_batch_job/__main__.py [hunk 2]
```diff
@@ -307,7 +308,9 @@ def process_coverage(args, job):
             '--object-directory=' + package_build_path,
             '-k',
             '-r', os.path.abspath('.'),
-            '--xml', '--output=' + outfile,
+            '--xml', '--output=' + outfile,         
+            '--gcov-exclude=.*#tests?#.*',
+            '--gcov-exclude=.*#gtest_vendor#.*',
```
Comment (nuclearsandwich) [inline ros2_batch_job/__main__.py:313]: Actually something else occurred to me. Are there implicit anchor metacharacters around these expressions that make the `.*` wrappers necessary? If not couldn't these be simplified to `#tests?#` and `#gtest_vendor#` respectively?

Comment (j-rivero) [reply ros2_batch_job/__main__.py:313]: The gcov generated files are in the form:
 ```
install#gtest_vendor#src#gtest_vendor#include#gtest#gtest.h.gcov                    src#ros2#rcl_logging#rcl_logging_spdlog#src#rcl_logging_spdlog.cpp.gcov
install#gtest_vendor#src#gtest_vendor#include#gtest#gtest-printers.h.gcov           src#ros2#rcl_logging#rcl_logging_spdlog#test#test_logging_interface.cpp.gcov
install#gtest_vendor#src#gtest_vendor#include#gtest#internal#gtest-internal.h.gcov
```
My local testing suggest that we can not remove the `.*`:

```
jrivero@shappire coverage_ws $ gcovr --verbose --gcov-exclude='.*#gtest_vendor#.*' --object-directory=build/rcl_logging_spdlog -k -r /home/jrivero/code/ros2/coverage_ws  -g | grep Excluding
Excluding gcov file: /home/jrivero/code/ros2/coverage_ws/install#gtest_vendor#src#gtest_vendor#include#gtest#gtest.h.gcov
Excluding gcov file: /home/jrivero/code/ros2/coverage_ws/install#gtest_vendor#src#gtest_vendor#include#gtest#internal#gtest-internal.h.gcov
Excluding gcov file: /home/jrivero/code/ros2/coverage_ws/install#gtest_vendor#src#gtest_vendor#include#gtest#gtest-printers.h.gcov
jrivero@shappire coverage_ws $ gcovr --verbose --gcov-exclude='#gtest_vendor#' --object-directory=build/rcl_logging_spdlog -k -r /home/jrivero/code/ros2/coverage_ws  -g | grep Excludingjrivero@shappire coverage_ws $ 
```

**Pull request merged**